tket/src/Converters/ZXConverters.cpp

```diff
@@ -0,0 +1,451 @@
+#include "Circuit/CircPool.hpp"
+#include "ZX/ZXDiagram.hpp"
+
+namespace tket {
+
+using namespace zx;
+
+enum class PortType { In, Out };
+typedef std::pair<VertPort, PortType> TypedVertPort;
+typedef boost::bimap<ZXVert, Vertex> BoundaryVertMap;
+
+bool is_spiderless_optype(const OpType& optype) {
+  return optype == OpType::Barrier || optype == OpType::SWAP ||
+         optype == OpType::noop;
+}
+
+// Add a swicth with the on-state controlled by the on_value.
+// qtype indicates the quantum type of the switch.
+std::pair<ZXVert, ZXVert> add_switch(
+    ZXDiagram& zxd, const bool& on_value, const QuantumType& qtype) {
+  zxd.multiply_scalar(std::sqrt(2.));
+  ZXVert triangle = zxd.add_vertex(ZXType::Triangle, qtype);
+  ZXVert x = zxd.add_vertex(ZXType::XSpider, 0, qtype);
+  zxd.add_wire(triangle, x, ZXWireType::Basic, qtype, 1);
+  if (on_value) {
+    // Turn on the switch if the input is 1
+    ZXVert negate = zxd.add_vertex(ZXType::XSpider, 1, QuantumType::Classical);
```

Comment:
This was marked as resolved without explanation. If `qtype` is set to `QuantumType::Quantum`, then you need this negate to flip both the positive and conjugated copies in the doubled-diagram picture, meaning it needs to also be doubled, right? If you plug in the constant 0 state as the control input, then it will merge with this spider and give a pi-phase bastard spider, so the final diagram will not be pure quantum as we would expect